infrastructure/Heroku.md

```diff
@@ -0,0 +1,127 @@
+# Heroku Hosting Guideline
+
+## Intro
+
+Heroku is the PaaS platform of choice for Rootstrap, even as, in order to support larger projects and offer greater value by reducing our customer's infrastructure costs we are driven to adopt larger cloud providers, primarily AWS in its multiple offerings: plain IaaS (EC2), PaaS (Elastic Beanstalk), or container-based platforms (ECS/EKS).
+This guide establishes a criteria for choosing Heroku as hosting platform for new projects and overall best practices around it.
+
+### Table of Contents 
+
+  - [Intro](#intro)
+    - [Pros](#pros)
+    - [Cons](#cons)
+  - [Heroku vs AWS](#heroku-vs-aws)
+    - [When to use Heroku](#when-to-use-heroku)
+    - [When to move to AWS](#when-to-move-to-aws)
+    - [Comparison Tools](#estimation-and-comparison-tools)
+  - [Best Practices](#best-practices)
+
+
+### Pros
+* Very easy to get started 
+* Free tier extends including most AddOns supports cheap PoC and MVP environments
+* Quick fixes based on customer feedback are deployed immediately
+* Setup can be implemented End2End without Linux Admin or DevOps expertise
+
+### Cons
+* Cost rise quickly after basic usage tiers
+* No automated scalability out of the box (Addon solutions are also expensive)
+* Not really feasible for very compute-intensive projects
+* Deployments can be slow for larger apps
+* Not all tech stacks supported
+
+
+## Heroku vs AWS
+
+### When to use Heroku
+
+In order to deploy a new project into Heroku, most of these conditions should be met:
+
+* Initial scope of the project is just an MVP
+* Project will have low resource requirements (RAM, CPU) even in Production short/medium-term
+* The project is built on a well-known and fully supported stack, eg.
+    * Language:
+        * Python: Django/Flask
+        * Ruby: Rails 4+
+        * Node.js 12+
+    * Database:
+        * PostgreSQL 11+
+    * In-memory data store:
+        * Redis 5
+    * Email:
+        * Sendgrid
+        * Mailgun
+    * Logging:
+        * Papertrail
+* There are no DevOps hours allocated to the project
+* There is no available [Reference Architecture](https://www.notion.so/rootstrap/Reference-Architectures) for AWS that fits the needs of the project
+
+### When to move to AWS 
+
+As a rule of thumb, when total ownership costs for the project in Heroku exceeds or are expected to exceed $350/month, it is worthwhile to invest in migrating to AWS.
```

Comment:
Out of curiosity, I believe that this number ($350) comes up from comparing AWS costs vs Heroku costs. But are you including the devops costs on the AWS side? It's hard to think of a devops charging less than $350 per month 🤷🏻‍♂️

Reply:
You don't hire a full time DevOps for a single $350/m system :)

The assumption here is: you invest in DevOps hours to build a deployment pipeline for the system in AWS such that configuring, deploying and testing is equally simple as in Heroku. 

This is a one-time cost (which in fact should decrease with each new project as code is reused); and from that point forward maintenance costs in man/hours are similar (admin costs around Heroku are low but not 0, only now they are mostly spread around Devs)

Reply:
Niceee :)